### ISSUE TITLE
Move on from Drupal VM?

### ISSUE THREAD

**Issue by pyrello** [uiowa/uiowa#3641]
There is a conversation going on about the future of Drupal VM: https://github.com/geerlingguy/drupal-vm/issues/2164

It doesn't seem, based on the conversation so far, that there is a clear path forward for Drupal VM. Jeff Geerling has committed to only minimal maintenance until he can't handle that anymore and then he's going to deprecate it.

While moving to a different virtualization paradigm is not a small thing, it is worth noting that we collectively spent a lot of hours resolving issues related to our current solution and figuring out how to continue using it. The introduction of the M1 Macs also raises questions about whether this is still the right solution for us.

Lando is an obvious alternative both because we have used it before and I think that docker has already figured out M1 compatibility.

Just wanted to have a place to keep the discussion rolling.

## Proposed Solution

- [x] Survey the options that provide drupal/acquia blt integration for M1 macs - https://github.com/uiowa/uiowa/issues/3641#issuecomment-970403213

Not that Slack support is key, but the current Drupal slack community numbers are ddev (801), lando (464), docksal (166). Releases also fall in that order ddev (146), lando (141), docksal (45). All seem to be releasing at a pretty regular schedule.

Lando pulls ahead in collaborators (193), ddev (159), docksal (62).

## Next Steps

- [x] Establish an experimental PR for ddev - https://github.com/uiowa/uiowa/pull/4458
- [x] Establish an experimental PR for lando - https://github.com/uiowa/uiowa/pull/4470
- [x] Try docksal
- [x] Test solutions on older/newer (M1) macs
- [x] Collectively decide on path forward?

**Comment by lexsoft00** [comment]
I'm sad to see DrupalVM go, it's been lovely to work with the past 3 years. 

My environment is on Windows 10 and the best alternative I'm working with right now is Windows WSL2 ( https://docs.microsoft.com/en-us/windows/wsl/install-win10 ) with Ubuntu 18.04 combined with Docker Desktop and DDEV ( https://ddev.readthedocs.io/en/stable/#installation-or-upgrade-windows-wsl2 ). 

You can easily mount a network drive from the `\\wsl$\ ` and use that path for IntelliJ Idea or PHPStorm and following Susi's blog ( https://susi.dev/fully-integrate-ddev-and-phpstorm-including-unit-tests-with-coverage-update-2021 ) you can have a fast and good integration.

I did not expect this to work so great but it's really easy and fast once the setup is done.

**Comment by kartagis** [comment]
I'm on a Mac and I have both lando and DrupalVM. These days I tend to use DrupalVM because lando is so slow with everything (drush operations, website surfing) on my system. Drush operations can be faster though, if I ssh into the container. I agree with @pyrello though on lando being a viable alternative, considering it doesn't add another 4~ GB container to the filesystem.

**Comment by pyrello** [comment]
Just noticed that Lando introduced experimental support for Apple Silicon: https://github.com/lando/lando/releases/tag/v3.3.0

**Comment by joewhitsitt** [comment]
DDEV supports M1 as well: https://ddev.readthedocs.io/en/stable/

**Comment by pyrello** [comment]
Here are some docs indicating how you can use DDEV and BLT together: https://dev.acquia.com/blog/blt-and-ddev-together

**Comment by joewhitsitt** [comment]
would be nice to have a spare computer to test with instead of risking our current machines

**Comment by joewhitsitt** [comment]
https://blog.lando.dev/2021/03/03/acquia-lando-integration/

**Comment by pyrello** [comment]
> https://blog.lando.dev/2021/03/03/acquia-lando-integration/

For this to be desirable, Lando would need to perform significantly better than it does when I am using https://github.com/thinktandem/drupal-contributions, which is extremely slow and makes my computer hum whenever it is running.

**Comment by joewhitsitt** [comment]
i just started poking around again because edudu slack is talking about lando moving away from docker.

>I'd presume they are also interested in prospects for getting better performance out of alternatives when using Apple Silicon.

**Comment by pyrello** [comment]
It is also possible that the experience with Lando I've been having is somehow unique to that project.

**Comment by bspeare** [comment]
DDEV + Mutagen: https://twitter.com/stephencross/status/1456987443161305093

**Comment by mjoneill** [comment]
1) Provide a list of possible alternatives for M1 support
2) Check in with team, figure out next step(s)

**Comment by richardbporter** [comment]
We should ensure that BLT works as well. Probably worth checking Acquia/BLT docs.

**Comment by joewhitsitt** [comment]
Just setup ddev + mutagen fairly painlessly (updating docker took the longest) following [these steps](https://ddev.readthedocs.io/en/stable/users/cli-usage/#drupal-9-composer-setup-example) and modifying

`ddev config --project-type=drupal9 --docroot=web --create-docroot`

to include 

`--mutagen-enabled`

```
mydrupal % ddev drush status
 Drupal version   : 9.2.8                                      
 Site URI         : https://mydrupal.ddev.site                 
 DB driver        : mysql                                      
 DB hostname      : ddev-mydrupal-db                           
 DB port          : 3306                                       
 DB username      : db                                         
 DB name          : db                                         
 Database         : Connected                                  
 Drupal bootstrap : Successful                                 
 Default theme    : bartik                                     
 Admin theme      : seven                                      
 PHP binary       : /usr/bin/php7.4                            
 PHP config       : /etc/php/7.4/cli/php.ini                   
 PHP OS           : Linux                                      
 Drush script     : /var/www/html/vendor/drush/drush/drush     
 Drush version    : 10.6.1                                     
 Drush temp       : /tmp                                       
 Drush configs    : /var/www/html/vendor/drush/drush/drush.yml 
 Install profile  : standard                                   
 Drupal root      : /var/www/html/web                          
 Site path        : sites/default                              
 Files, Public    : sites/default/files                        
 Files, Temp      : /tmp  
 ```

Caveats about mutagen - https://ddev.readthedocs.io/en/stable/users/performance/#caveats-about-mutagen-integration

Update: mutagen no longer? https://github.com/drud/ddev/issues/2278, https://github.com/lando/lando/issues/2354

Multi-site:
https://github.com/drud/ddev-contrib/blob/master/recipes/drupal8-multisite/README.md

Acquia/Uiowa:
https://ddev.readthedocs.io/en/stable/users/providers/acquia/

**Comment by joewhitsitt** [comment]
Progress:

<img width="636" alt="Screen Shot 2021-11-15 at 4 40 38 PM" src="https://user-images.githubusercontent.com/4663676/141864174-d5b27b1a-4d17-4582-8045-1fd3bf3ed385.png">

**Comment by joewhitsitt** [comment]
To answer the issue's question:

I believe to support M1 macs, we are looking at docker for mac based solutions. Docksal, Lando, DDEV. They all support Acquia BLT. Lando and DDEV both have issues to get away from Docker licensing by providing an alternative foundation but I think that is still just in the discussion phase.

Lando and DDEV both have experimental support for Mutagen which syncs (doubles) container assets for faster on the fly development, but that is being rolled back: drud/ddev#2278, lando/lando#2354

We still have a lot of our lando footprint still available in the repo, so in terms of adoption, we probably have a faster path there. Older and newer macs are the big question there along with any other "not ideals" that are still present from before.

I am impressed with the speed and ease to get DDEV up in running in some capacity but more work needs to be done to get multisite support to where it is currently

**Comment by joewhitsitt** [comment]
This looks interesting... https://ddev.com/ddev-local/sharing-a-ddev-local-project-with-other-collaborators/

**Comment by pyrello** [comment]
To test out lando, I went back to our old `.lando.yml` setup: https://github.com/uiowa/uiowa/commit/14d3704d2c822b8b78603c61d9a710ea48c69d99

It needs some tinkering to get it to work (ahem @richardbporter), but when running the `lando rebuild -y` command, it is significantly faster than when using the `drupal-contributions` project. My computer is also not humming as much, although it is humming a little bit.

I can't remember what our exact performance bottlenecks were, so not sure how to test it beyond this.

**Comment by joewhitsitt** [comment]
Started a DDEV PR that folks can try elsewhere on their computer. https://github.com/uiowa/uiowa/pull/4458

**Comment by joewhitsitt** [comment]
I can't seem to get Lando to a working point with multisite and the frontend stuff (node/nvm/yarn).

ssh keys seem to work with composer install as it relates to uiowa_auth (private repo), but not with `blt ds`. Lando says they are auto passed in and I see it in the logs but it still doesn't seem to be happy.

**Comment by richardbporter** [comment]
> I can't seem to get Lando to a working point with multisite and the frontend stuff (node/nvm/yarn).
> 
> ssh keys seem to work with composer install as it relates to uiowa_auth (private repo), but not with `blt ds`. Lando says they are auto passed in and I see it in the logs but it still doesn't seem to be happy.

Having the same issue.

DDev is working really well for me. I wonder if other people can try it to see if they have any issues with it. That might inform whether we put more effort into #4470 or:

>Establish an experimental PR for docksal?

DDev is working with @e-marie-w's M1.

**Comment by joewhitsitt** [comment]
>Establish an experimental PR for docksal?

Based on the usage/support I have seen, I am less excited about docksal.

**Comment by richardbporter** [comment]
Some ddev/lando thoughts:

- ddev auth ssh asks for SSH key password once and then SSH commands just work after that. Lando asks every time - might require additional work. (`https://github.com/lando/lando/issues/478`)
- ddev loads SSL cert into system so browsers trust it. Lando doesn't.
- ddev ships with yarn and has special integration with it for more performant FS changes. Lando doesn't.
- ddev ships with NVM, lando doesn't.
- lando has amazing CLI messages 💯, ddev doesn't.
- lando syncs .git directory by default, ddev doesn't but can, perhaps at a performance hit.
- ddev seems faster (no real data to back this up) but could be related to above.
- lando enables xdebug when the containers are built, ddev doesn't but can enable/disable dynamically with a CLI command. This can improve performance.
- lando has a specific Acquia recipe.

**Comment by richardbporter** [comment]
Trying to install Docksal, I get: 
```
 ERROR:  Failed starting the proxy service.
 ERROR:  Updates failed. fin was not updated.
```

Initializing a docksal project results it:
```
ERROR: yaml.reader.ReaderError: unacceptable character #x001b: special characters are not allowed
  in "~/uiowa/.docksal/docksal.yml", position 0
 NOTICE:  Some Docksal services were not running
         Restarting Docksal system services...
```
for a file that it generated. Fixing up the file and then restarting resulted in even more errors. Tried setting up an acquia project, got more errors.

Can we scratch docksal?

**Comment by joewhitsitt** [comment]
>Can we scratch docksal?

+1

**Comment by joewhitsitt** [comment]
>lando has amazing CLI messages 💯, ddev doesn't.

This isn't weighted high enough IMO.

Maybe we can just use these as inspiration when crafting our custom commands
